### PR TITLE
Key Vault: Converting from *Base to *Properties

### PR DESCRIPTION
**Pull request by iscai-msft**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py [hunk 1]
```diff
@@ -15,11 +15,11 @@
     from ._shared._generated.v7_0 import models as _models
 
 
-class SecretAttributes(object):
+class SecretProperties(object):
     """A secret's id and attributes."""
 
     def __init__(self, attributes, vault_id, **kwargs):
-        # type: (_models.SecretAttributes, str, **Any) -> None
+        # type: (_models.SecretProperties, str, **Any) -> None
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py]: Has the generated class's name changed?

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py [hunk 2]
```diff
@@ -163,27 +163,39 @@ def tags(self):
...
         """Construct a Secret from an autorest-generated SecretBundle"""
         return cls(
-            secret_bundle.attributes,
-            secret_bundle.id,
-            secret_bundle.value,
-            content_type=secret_bundle.content_type,
-            key_id=secret_bundle.kid,
-            managed=secret_bundle.managed,
-            tags=secret_bundle.tags,
+            properties=SecretProperties(
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py]: Could instead call `SecretProperties._from_secret_bundle`.

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py [hunk 3]
```diff
@@ -163,27 +163,39 @@ def tags(self):
...
+                tags=secret_bundle.tags
+            ),
+            value=secret_bundle.value
         )
 
+    @property
+    def properties(self):
+        # type: () -> SecretProperties
+        """
+        The secret properties
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py]: Implies the properties are themselves secret. You probably want "the secret's properties".

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py [hunk 4]
```diff
@@ -163,27 +163,39 @@ def tags(self):
         return self._tags
 
 
-class Secret(SecretAttributes):
-    """All a secret's attributes, and its value."""
+class Secret(object):
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py]: We decided all models should have accessors for identifying properties i.e. id, name, vault url and version.

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/client.py [hunk 5]
```diff
@@ -169,16 +169,16 @@ def update_secret(
             error_map={404: ResourceNotFoundError},
             **kwargs
         )
-        return SecretAttributes._from_secret_bundle(bundle)  # pylint: disable=protected-access
+        return SecretProperties._from_secret_bundle(bundle)  # pylint: disable=protected-access
 
     @distributed_trace
-    def list_secrets(self, **kwargs):
-        # type: (**Any) -> ItemPaged[SecretAttributes]
+    def list_secret_properties(self, **kwargs):
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/client.py]: This should still be `list_secrets`. `list_secret_properties` suggests listing the properties of a single secret; actually it lists all secrets in the vault and happens to return a class we've named `SecretProperties`.

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/aio/client.py [hunk 6]
```diff
@@ -147,15 +147,15 @@ class SecretClient(AsyncKeyVaultClientBase):
             error_map={404: ResourceNotFoundError},
             **kwargs
         )
-        return SecretAttributes._from_secret_bundle(bundle)  # pylint: disable=protected-access
+        return SecretProperties._from_secret_bundle(bundle)  # pylint: disable=protected-access
 
     @distributed_trace
-    def list_secrets(self, **kwargs: "**Any") -> AsyncIterable[SecretAttributes]:
+    def list_secret_properties(self, **kwargs: "**Any") -> AsyncIterable[SecretProperties]:
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/aio/client.py]: Similarly, this should remain `list_secrets`

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py (hunk 1 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py]: thanks, I got a little too trigger happy

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py (hunk 2 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py]: changing, thank you!

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py (hunk 3 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py]: makes sense, switching the comment

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/aio/client.py (hunk 6 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/aio/client.py]: switched back to list_secrets, thanks!

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/client.py (hunk 5 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/client.py]: switched back to list_secrets, thanks!

File: sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py (hunk 4 above)
Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/models.py]: added those accessors to Secret and DeletedSercret, thanks!

**Pull request merged**